### PR TITLE
Boilerplate tests

### PR DESCRIPTION
**Pull request by osdiab**
Add testing examples using enzyme. Yay :)

### CODE REVIEW

File: react-redux-boilerplate/test/components/QuoteSection_test.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    const quoteSectionWithQuote = shallow(
+      <QuoteSectionView fetchingQuote={false} fetchQuote={() => {}} fetchError={{ error: true }}
+        author={author} quote={quote}
+      />
+    );
+    assert.equal(quoteSectionWithQuote.find('.QuoteSection--welcome').length, 0);
+    assert.equal(quoteSectionWithQuote.find('.QuoteSection--loading').length, 0);
+    assert.equal(quoteSectionWithQuote.find('.QuoteSection--error').length, 1);
+    assert.equal(quoteSectionWithQuote.find('.QuoteSection--author').length, 0);
+    assert.equal(quoteSectionWithQuote.find('.QuoteSection--quote').length, 0);
```
Comment (nathanleiby) [inline react-redux-boilerplate/test/components/QuoteSection_test.jsx]: these look like things that would be great to write via table driven tests

**Comment by nathanleiby** [comment]
💯

File: react-redux-boilerplate/test/components/QuoteSection_test.jsx (hunk 1 above)
Comment (osdiab) [inline react-redux-boilerplate/test/components/QuoteSection_test.jsx]: be01dcc

File: react-redux-boilerplate/test/components/QuoteSection_test.jsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      loading: 0,
+      error: 1,
+      author: 0,
+      quote: 0,
+    };
+
+    for (const element in expectedLengths) {
+      if (!expectedLengths.hasOwnProperty(element)) {
+        continue;
+      }
```
Comment (nathanleiby) [inline react-redux-boilerplate/test/components/QuoteSection_test.jsx]: nit: probably an array of dicts would be easier to work with?

```
const testSpecs = [
  { element: "welcome", length: 0},
  { element: "quote", length: 1, text: "somequote"},
  ...
]
```

Comment (nathanleiby) [inline react-redux-boilerplate/test/components/QuoteSection_test.jsx]: Then I think you probably could move all of this to a helper method like `assertQuoteSection(spec)`

Comment (osdiab) [inline react-redux-boilerplate/test/components/QuoteSection_test.jsx]: this looking better? 5a14c5b

Comment (osdiab) [inline react-redux-boilerplate/test/components/QuoteSection_test.jsx]: Another iteration: ba966ab

File: react-redux-boilerplate/test/components/QuoteSection_test.jsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      assert.equal(loadingDom.text(), expectedText);
+    }
+  });
+
+  it('displays a loading message and not an error when both present', () => {
+    const quoteSection = shallow(
+      <QuoteSectionView fetchingQuote fetchQuote={() => {}} fetchError={{ error: true }} />);
+    assert.equal(quoteSection.find('.QuoteSection--welcome').length, 0);
+    assert.equal(quoteSection.find('.QuoteSection--error').length, 0);
+    assert.equal(quoteSection.find('.QuoteSection--loading').length, 1);
```
Comment (nathanleiby) [inline react-redux-boilerplate/test/components/QuoteSection_test.jsx]: could this use table test too?

**Comment by nathanleiby** [comment]
lgtm

File: react-redux-boilerplate/test/components/QuoteSection_test.jsx (hunk 3 above)
Comment (osdiab) [inline react-redux-boilerplate/test/components/QuoteSection_test.jsx]: dd961aa

**Pull request merged**